### PR TITLE
type_5_workload

### PR DESCRIPTION
**Pull request by kulrk**
Signed-off-by: kulrk <rohan.1.kulkarni@seagate.com>

## Problem Statement

-   Problem statement

## Design

-   For Bug, Describe the fix here.
-   For new test addition post the link for design

## Coding

   Checklist for Author

-   [ ]  Coding conventions are followed and code is formatted (e.g. using pycharm in-built formatter)

## Testing

  Checklist for Author

-   [ ]  New/Affected tests are executed
-   [ ]  Attach test execution logs

## Review 

  Checklist for Author

-   [ ]  JIRA number/GitHub Issue added to PR
-   [ ]  PR is self reviewed
-   [ ]  Jira and state/status is updated and JIRA is updated with PR link
-   [ ]  Check if the description is clear and explained

## Documentation

  Checklist for Author

-   [ ]  Changes done to ReadMe

### CODE REVIEW

File: workload/s3/IOStability/type_5_bucket.yaml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  TEST_ID: #TBD
+  object_size:
+    start: 0b
+    end: 512Mb
+  min_runtime: 2h
+  sessions_per_node: 5
+  number_of_buckets: 100
+  tool: s3api
+  operation: type_5_bucket_ops
+test_10:
```
Comment (DhobaleVishal) [inline workload/s3/IOStability/type_5_bucket.yaml]: Keep blank space between two tests.

File: workload/s3/IOStability/type_5_bucket.yaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  object_size:
+    start: 0b
+    end: 512Mb
+  min_runtime: 2h
+  sessions_per_node: 5
+  number_of_buckets: 100
+  tool: s3api
+  operation: type_5_bucket_ops
+test_9:
+  TEST_ID: #TBD
```
Comment (DhobaleVishal) [inline workload/s3/IOStability/type_5_bucket.yaml]: Instread TBD use TEST-XXXX for time being.

File: workload/s3/IOStability/type_5_negative.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  operation: type_5_object
+test_3: #negative scenario
+  TEST_ID: #TBD
+  object_size:
+    start: 0Kb
+    end: 5Gb
+  number_of_buckets: 1
+  number_of_objects: 500
+  operation: type_5_object_range_read
+test_4: #negative scenario
```
Comment (DhobaleVishal) [inline workload/s3/IOStability/type_5_negative.yaml]: @VikasKumarC does negative scenarios planned to support in CORIO?

**Review by DhobaleVishal — CHANGES_REQUESTED** [review]

File: workload/s3/IOStability/type_5_negative.yaml (hunk 3 above)
Comment (VikasKumarC) [reply workload/s3/IOStability/type_5_negative.yaml]: @ketanarlulkar  negative scenarios  are not in scope of  CORIO.

Comment (kulrk) [reply workload/s3/IOStability/type_5_negative.yaml]: @VikasKumarC , @DhobaleVishal  Type-5 includes negative scenarios.

**Review by VikasKumarC — APPROVED** [review]

**Review by DhobaleVishal — APPROVED** [review]

File: src/commons/utils/s3_utils.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    first letter should be a number or lowercase letter
+    rest letters can include number, lowercase, hyphens and dots.
+    bucket length can vary from 3 to 63
+    """
+    first = ''.join(random.SystemRandom().choice(string.ascii_lowercase + string.digits))
+    bkt_len = random.randint(2, 63)
+    rest = first + ''.join(random.SystemRandom().choice(string.ascii_lowercase + string.digits +
+                                                        "." + '-') for _ in range(bkt_len))
+    bucket_name = first + rest
+    return bucket_name
```
Comment (DhobaleVishal) [inline src/commons/utils/s3_utils.py]: optimize this lines.

Comment (kulrk) [reply src/commons/utils/s3_utils.py]: fixed

**Review by DhobaleVishal — APPROVED** [review]

**Pull request merged**